unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs

```diff
@@ -126,6 +140,19 @@ public void Populate(Model chatEntryModel)
             body.text = $"{chatEntryModel.bodyText}";
         }
 
+        if (CoordinateUtils.HasValidTextCoordinates(body.text)) {
+            CoordinateUtils.GetTextCoordinates(body.text).ForEach(c=> {
```

Comment:
LinQ's ForEach obscure a bit the code, I rather have the collection in a local variable and iterate on it.

Also, `foreach` suffers a small performance hit, when possible is preferable to use `for`. Since `CoordinateUtils.GetTextCoordinates` returns a List, you can apply it in this case.